### PR TITLE
Keyboard: Add a new keyboard MiniAxe.

### PR DESCRIPTION
**Pull request by ka2hiro**
Add a new keyboard MiniAxe.
I'm planing to sell this keyboard kit at the end of this month.

### CODE REVIEW

File: keyboards/miniaxe/readme.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+MiniAxe
+===
+
+![MiniAxe](https://i.imgur.com/1ApzrCz.jpg)
+
+Yet another split ortholinear keyboard with 3x5+3 keys.
+
+Keyboard Maintainer: [ka2hiro](https://github.com/ka2hiro) [@ka2hiro](https://twitter.com/ka2hiro)
```
Comment (noroadsleft) [inline keyboards/miniaxe/readme.md]: This line needs another space at the end for the file to be rendered as intended.

```suggestion
Keyboard Maintainer: [ka2hiro](https://github.com/ka2hiro) [@ka2hiro](https://twitter.com/ka2hiro)  
```

File: keyboards/miniaxe/readme.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+MiniAxe
+===
+
+![MiniAxe](https://i.imgur.com/1ApzrCz.jpg)
+
+Yet another split ortholinear keyboard with 3x5+3 keys.
+
+Keyboard Maintainer: [ka2hiro](https://github.com/ka2hiro) [@ka2hiro](https://twitter.com/ka2hiro) 
+Hardware Supported: MiniAxe PCB, ATMEGA32U4
```
Comment (noroadsleft) [inline keyboards/miniaxe/readme.md]: This line needs two spaces at the end for the file to be rendered as intended.

```suggestion
Hardware Supported: MiniAxe PCB, ATMEGA32U4  
```

File: keyboards/miniaxe/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+#define _ADJUST 16
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  ADJUST,
+};
+
+#define _______ KC_TRNS
```
[the remark this answers is not in the document]
Comment (ka2hiro) [reply keyboards/miniaxe/keymaps/default/keymap.c]: Remove unnecessary keycode aliases.

File: keyboards/miniaxe/keymaps/underglow/keymap.c [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+#define _ADJUST 16
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  ADJUST,
+};
+
+#define _______ KC_TRNS
```
[the remark this answers is not in the document]
Comment (ka2hiro) [reply keyboards/miniaxe/keymaps/underglow/keymap.c]: Remove unnecessary keycode aliases.

File: keyboards/miniaxe/readme.md (hunk 1 above)
Comment (ka2hiro) [reply keyboards/miniaxe/readme.md]: Added a space.

File: keyboards/miniaxe/readme.md (hunk 2 above)
Comment (ka2hiro) [reply keyboards/miniaxe/readme.md]: Added two spaces. Thanks.

**Comment by drashna** [comment]
@ka2hiro it looks like the only difference here is the matrix code to accommodate for a very unusual diode configuration.

1. Is there a specific reason for this (more of curious)
2. Would you mind pulling the matrix changes OUT of this PR, and open a new PR to merge these into the Split Common code instead? (located in `/quantum/split_common/`

**Comment by ka2hiro** [comment]
> @ka2hiro it looks like the only difference here is the matrix code to accommodate for a very unusual diode configuration.
> 
> 1. Is there a specific reason for this (more of curious)
This keyboard has no diodes. Key switches are directly connected to MPU's pins, which means 0 rows/18 cols. It does't fit to existing matrix.c.  So, I decided to create own (modified) matrix.c.

> 2. Would you mind pulling the matrix changes OUT of this PR, and open a new PR to merge these into the Split Common code instead? (located in `/quantum/split_common/
I think this design (no diode) isn't common, so it is better to separate it from split common code.

File: keyboards/miniaxe/info.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "MiniAxe",
+    "url": "",
+    "maintainer": "ka2hiro",
+    "bootloader": "atmel-dfu",
```
[the remark this answers is not in the document]
Comment (ka2hiro) [reply keyboards/miniaxe/info.json]: I removed it. Thanks.

**Comment by drashna** [comment]
Oh, okay, thanks for clarifying that then!

**Comment by ka2hiro** [comment]
@drashna Thank you for your understanding!

**Comment by drashna** [comment]
You're very welcome! 

Ideally, I want to use the Split Common code as much as possible, and remove most (if not all) boards that have the code localized.  That way, if there are changes, it's applied globally, not just to one keyboard. 

That was what my concern was here.  But you're using the split common with a custom matrix for a very unusual config.  So ... I feel okay about how it's implemented.  Though, you may want to keep an eye on changes to the split matrix, just in case.

**Pull request merged**